### PR TITLE
support lazy html imports

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

**Comment by garlicnation** [comment]
@justinfagnani About to push the manual lazy feature inclusion we discussed over VC earlier.

File: src/html/html-import-scanner.ts [hunk 1]
```diff
@@ -22,9 +22,28 @@ import {HtmlScanner} from './html-scanner';
...
+    notCssLink,
+    p.parentMatches(
+      p.AND(
+        p.hasTagName('dom-module'),
+        p.NOT(p.hasTagName('template'))
+      )
+    )
+  );
 
 export class HtmlImportScanner implements HtmlScanner {
```
Comment (justinfagnani) [inline src/html/html-import-scanner.ts]: Could you ad a comment that mentions lazy imports?

**Comment by justinfagnani** [comment]
I need to get used to the fact that some comments don't go out immediately, and not forget to send them :)

**Comment by justinfagnani** [comment]
This is LGTM, if you want to merge it first.

File: src/html/html-import-scanner.ts (hunk 1 above)
Comment (garlicnation) [inline src/html/html-import-scanner.ts]: Done.

File: src/html/html-import-scanner.ts [hunk 2]
```diff
@@ -22,9 +22,28 @@ import {HtmlScanner} from './html-scanner';
...
+    linkTag,
+    p.hasSpaceSeparatedAttrValue('rel', 'import'),
+    notCssLink,
+    p.NOT(
+      p.parentMatches(p.hasTagName('template')))
+);
+
+const isLazyImportNode = p.AND(
+    p.hasTagName('link'),
+    p.hasSpaceSeparatedAttrValue('rel', 'lazy-import'),
```
Comment (garlicnation) [inline src/html/html-import-scanner.ts]: Changed the logic - now it bails if it encounters that.

**Pull request merged**